### PR TITLE
Add start and end times to game log table

### PR DESCRIPTION
**Pull request by h-shan**
Remove 'Time to Boot' from table because it was made redundant by the
'Start Time' column.

Refactor functions to remove unnecessary subscripting.

Fix bug with games not recording finish time.

### CODE REVIEW

File: wlct/cogs/clot.py [hunk 1]
```diff
@@ -26,7 +26,6 @@ def __init__(self, bot):
                     print("Saving discord id: {} for user".format(ctx.message.author.id))
                     discord_obj = DiscordUser(memberid=ctx.message.author.id)
                     discord_obj.save()
-                    player.discord = discord_obj
```
Comment (brendanfly) [inline wlct/cogs/clot.py]: This needs to be player.discord_member now in order to link properly.

File: wlct/tournaments.py [hunk 2]
```diff
@@ -2422,12 +2422,19 @@ def finish_game(self):
         self.is_finished = True
 
         self.current_state = "Finished"
+        self.tournament = find_tournament_by_id(self.tournament.id)
```
Comment (brendanfly) [inline wlct/tournaments.py]: please create a swiss or seeded tournament and hit start, make sure it looks good at the end with no crashes.

File: wlct/tournaments.py [hunk 3]
```diff
@@ -2422,12 +2422,19 @@ def finish_game(self):
...
         log_tournament("Finish Game/Game Entry: {}".format(self.teams), self.tournament)
         team1 = self.teams.split('.')[0]
         team2 = self.teams.split('.')[1]
 
         tournament_team1 = get_team_by_id(self.tournament, int(team1))
         tournament_team2 = get_team_by_id(self.tournament, int(team2))
+
```
Comment (brendanfly) [inline wlct/tournaments.py]: Please add a comment...."if this tournament is actually a sub-tournament, look it up like this"

File: wlct/tournaments.py [hunk 4]
```diff
@@ -2448,6 +2455,9 @@ def finish_game(self):
             self.game_finished_time = timezone.now()
             self.save()
 
+        else:
+            log('Could not find tournament teams with id\'s: {} {} in tournament {}'.format(team1, team2, self.tournament.id), LogLevel.error)
```
Comment (brendanfly) [inline wlct/tournaments.py]: please use log_tournament, which takes the actual tournament in as an arg too for better logging.

File: wlct/tournaments.py [hunk 5]
```diff
@@ -3838,36 +3850,33 @@ def update_game_log(self):
...
-                log += '</tr>'
+                
+                start_time = game.game_start_time.strftime("%b %d, %Y %H:%M:%S %p")
 
-        log += '</tbody></table>'
-        self.game_log = log
+                # game.game_finished_time check is redundant but is done for backwards compability with bad existing data
+                end_time = game.game_finished_time.strftime("%b %d, %Y %H:%M:%S %p") if game.is_finished and game.game_finished_time else 'N/A'
+
+                game_log += '<td>{}</td>'.format(start_time)
```
Comment (brendanfly) [inline wlct/tournaments.py]: Am I crazy or did you invent another column here?

File: wltourney/static/js/tournament.js [hunk 6]
```diff
@@ -578,10 +578,6 @@ $(function () {
                           {
                                $("#gamelogtab").html(data.game_log);
                                $('#game_log_data_table').DataTable();
-
-                               $('.time_to_boot').each(function () {
```
Comment (brendanfly) [inline wltourney/static/js/tournament.js]: for my education what's the reason this isn't needed?

File: wlct/tournaments.py (hunk 5 above)
Comment (h-shan) [reply wlct/tournaments.py]: Yea I added two columns to the game log (Start Time and End Time) and removed the Time to boot column.

File: wltourney/static/js/tournament.js (hunk 6 above)
Comment (h-shan) [reply wltourney/static/js/tournament.js]: I removed the time to boot column.

File: wlct/tournaments.py (hunk 5 above)
Comment (brendanfly) [reply wlct/tournaments.py]: You will also need to add in another <th> tag at the top I thought. Does this render properly?

File: wltourney/static/js/tournament.js (hunk 6 above)
Comment (brendanfly) [reply wltourney/static/js/tournament.js]: Nice.

File: wlct/cogs/clot.py (hunk 1 above)
Comment (h-shan) [reply wlct/cogs/clot.py]: Done

File: wlct/tournaments.py (hunk 2 above)
Comment (h-shan) [reply wlct/tournaments.py]: No crashes, but some columns seem to be unimplemented:

- Swiss tournament
  - Tournament game log
- Seeded tournament
  - Tournament game log
  - Bracket/Games

File: wlct/tournaments.py (hunk 3 above)
Comment (h-shan) [reply wlct/tournaments.py]: Done

File: wlct/tournaments.py (hunk 2 above)
Comment (brendanfly) [reply wlct/tournaments.py]: Columns in the database? Or on the actual page?

File: wlct/tournaments.py (hunk 4 above)
Comment (h-shan) [reply wlct/tournaments.py]: Done

File: wlct/tournaments.py (hunk 5 above)
Comment (h-shan) [reply wlct/tournaments.py]: I did add another column in the top.

File: wlct/tournaments.py (hunk 2 above)
Comment (h-shan) [reply wlct/tournaments.py]: On the actual page, the tabs are blank

Comment (brendanfly) [reply wlct/tournaments.py]: For swiss I think that's OK, for seeded the bracket should be present. That is probably another bug...ugh...ok I will take a look at this tomorrow then but it sounds like there are some issues generating this data then.

File: wlct/tournaments.py [hunk 7]
```diff
@@ -2422,12 +2422,20 @@ def finish_game(self):
         self.is_finished = True
 
         self.current_state = "Finished"
+        self.tournament = find_tournament_by_id(self.tournament.id)
```
Comment (brendanfly) [inline wlct/tournaments.py]: Actually on second thought we should never overwrite the tournament member here. I'm going to debug this today and see why it's not able to find it.

**Pull request merged**